scripts/elasticsearch_nodes.py

```diff
@@ -0,0 +1,130 @@
+"""
+This script is referenced by our TerraForm config as an external data source. It
+prints the current deployment's Elasticsearch node IDs as JSON to STDOUT. If the
+list of node IDs contain fewer items than the number of placeholders in the
+Cloudwatch dashboard template, the output is padded with 'blank' entries.
+
+Note that Terraform requires the object given to an external data source is a
+map of string values, so the list of node IDs is JSON encoded into a string and
+later converted back into a list when processed in the Terraform config.
+
+Example output when using a template that has four node ID placeholders and a
+current deployment that has two ES nodes:
+{"nodes": "[\"tVsGH2BSXXYe9sMLEYRcIv\", \"ydG9meZMAGkTnAlPgLljz8\", \"blank\", \"blank\"]"}
+"""
+from itertools import (
+    repeat,
+)
+import json
+import logging
+from time import (
+    sleep,
+)
+from typing import (
+    Sequence,
+)
+
+from botocore.exceptions import (
+    ClientError,
+)
+import jq
+
+from azul import (
+    config,
+    require,
+)
+from azul.deployment import (
+    aws,
+)
+from azul.logging import (
+    configure_script_logging,
+)
+
+log = logging.getLogger(__name__)
+
+# How many times to try getting the node IDs from AWS before giving up.
+max_attempts = 10
+
+
+def main() -> None:
+    node_ids = get_node_ids()
+    data = prepare_node_ids(node_ids)
+    print(json.dumps(data))
+
+
+def get_node_ids() -> Sequence[str]:
+    """
+    Return a list of ES node IDs used by the current deployment.
+    """
+    for i in range(max_attempts):
+        log.debug(f'Attempt {i}/{max_attempts}')
+        try:
+            response = aws.cloudwatch.list_metrics(Namespace='AWS/ES',
+                                                   MetricName='CPUUtilization',
+                                                   RecentlyActive='PT3H',
+                                                   Dimensions=[
+                                                       {
+                                                           'Name': 'DomainName',
+                                                           'Value': config.es_domain
+                                                       }
+                                                   ])
+        except ClientError as e:
+            log.warning(e.response['Error']['Code'] + '. Retrying...')
```

Comment:
--

Reply:
Is a change requested here? Using this value produces a log message such as:
```
WARNING MainThread: InvalidParameterValue. Retrying...
```